src/open-id-connect.ts

```diff
@@ -26,7 +26,10 @@ export default class OpenIdConnect {
   }
 
   public async login(): Promise<void> {
-    const args: any = {};
+    const instruction = this.router.currentInstruction;
+    const redirectUrl = instruction.queryParams.loginRedirectRoute || this.getInstructionUrl(instruction);
+
+    const args: any = {data: redirectUrl};
```

Comment:
Still requires setting the data param for the app redirect after sign-in.
We need the additional state param added to the signinRedirect.
The redirect_url is the one which was configured on the security token service (STS) server.

Reply:
Let me see if I understand correctly. We cannot pass a different `redirect_url` here, because the `redirect_url` has to match what we've configured at the authorization server. Is that right?

Reply:
some servers require that they match
some allow passing an url to redirect to..

Reply:
Yes, it has to be an exact match of a registered URL.
Without it would be a security risk where an attacker can use the client_id and its own redirect url.
That's why the state param must be used.